opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -231,6 +234,9 @@ def detect(self) -> "Resource":
                     item.split("=") for item in env_resources_items.split(",")
                 )
             }
+        service_name = os.environ.get(OTEL_SERVICE_NAME)
+        if service_name:
+            env_resource_map["service.name"] = service_name
```

Comment:
Should this use the semconv package instead of the string for the key?

Reply:
Fixed.